File: edtc_matlab/ar1.m
% Filename: ar1.m
% Author: Andy Qi
% Date: December 2008
% Corresponds to: Listing 8.1

a = 0.5;
b = 1;
X = zeros(1, 101);    % Create an empty array to store path
X(1) = normrnd(0, 1); % X_0 has dist N(0, 1)
for t = 1:100
    X(t + 1) = normrnd(a * X(t) + b, 1);
end

File: python_code/ar1.py
# Filename: ar1.py
# Author: John Stachurski
# Date: December 2008
# Corresponds to: Listing 8.1

from random import normalvariate as N

a, b = 0.5, 1       # Parameters
X = {}              # An empty dictionary to store path
X[0] = N(0, 1)      # X_0 has distribution N(0, 1)

for t in range(100):
    X[t+1] = N(a * X[t] + b, 1)


